features/messages/impl/src/test/kotlin/io/element/android/features/messages/impl/utils/FakeTextPillificationHelper.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.impl.utils
+
+class FakeTextPillificationHelper(
+    private val pillifyLambda: (CharSequence) -> CharSequence = { it }
```

Comment:
I prefer to have a default lambda content with `lambdaError()` so that we ensure that it's not called when it's not expected, which is actually doing more test. Note that in this particular case, this is maybe OK to keep the current default implementation. I let you choose what's best.

Reply:
This component is sometimes used as a dependency deep in the hierarchy so I think it's better to not have to customise it for testing other components that may depend on it for functionality but don't check its output.